// google-ads-node/build/src/index.js
"use strict";
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionGoalCampaignConfigServiceClient = exports.ConversionCustomVariableServiceClient = exports.ConversionAdjustmentUploadServiceClient = exports.ConversionActionServiceClient = exports.CampaignSharedSetServiceClient = exports.CampaignServiceClient = exports.CampaignLabelServiceClient = exports.CampaignGroupServiceClient = exports.CampaignFeedServiceClient = exports.CampaignExtensionSettingServiceClient = exports.CampaignExperimentServiceClient = exports.CampaignDraftServiceClient = exports.CampaignCustomizerServiceClient = exports.CampaignCriterionServiceClient = exports.CampaignConversionGoalServiceClient = exports.CampaignBudgetServiceClient = exports.CampaignBidModifierServiceClient = exports.CampaignAssetSetServiceClient = exports.CampaignAssetServiceClient = exports.BillingSetupServiceClient = exports.BiddingStrategyServiceClient = exports.BiddingSeasonalityAdjustmentServiceClient = exports.BiddingDataExclusionServiceClient = exports.BatchJobServiceClient = exports.AudienceServiceClient = exports.AudienceInsightsServiceClient = exports.AssetSetServiceClient = exports.AssetSetAssetServiceClient = exports.AssetServiceClient = exports.AssetGroupSignalServiceClient = exports.AssetGroupServiceClient = exports.AssetGroupListingGroupFilterServiceClient = exports.AssetGroupAssetServiceClient = exports.AdServiceClient = exports.AdParameterServiceClient = exports.AdGroupServiceClient = exports.AdGroupLabelServiceClient = exports.AdGroupFeedServiceClient = exports.AdGroupExtensionSettingServiceClient = exports.AdGroupCustomizerServiceClient = exports.AdGroupCriterionServiceClient = exports.AdGroupCriterionLabelServiceClient = exports.AdGroupCriterionCustomizerServiceClient = exports.AdGroupBidModifierServiceClient = exports.AdGroupAssetServiceClient = exports.AdGroupAdServiceClient = exports.AdGroupAdLabelServiceClient = exports.AccountLinkServiceClient = exports.AccountBudgetProposalServiceClient = exports.v11 = void 0;
exports.SmartCampaignSettingServiceClient = exports.SharedSetServiceClient = exports.SharedCriterionServiceClient = exports.RemarketingActionServiceClient = exports.RecommendationServiceClient = exports.ReachPlanServiceClient = exports.PaymentsAccountServiceClient = exports.OfflineUserDataJobServiceClient = exports.MerchantCenterLinkServiceClient = exports.MediaFileServiceClient = exports.LabelServiceClient = exports.KeywordThemeConstantServiceClient = exports.KeywordPlanServiceClient = exports.KeywordPlanIdeaServiceClient = exports.KeywordPlanCampaignServiceClient = exports.KeywordPlanCampaignKeywordServiceClient = exports.KeywordPlanAdGroupServiceClient = exports.KeywordPlanAdGroupKeywordServiceClient = exports.InvoiceServiceClient = exports.GoogleAdsServiceClient = exports.GoogleAdsFieldServiceClient = exports.GeoTargetConstantServiceClient = exports.FeedServiceClient = exports.FeedMappingServiceClient = exports.FeedItemTargetServiceClient = exports.FeedItemSetServiceClient = exports.FeedItemSetLinkServiceClient = exports.FeedItemServiceClient = exports.ExtensionFeedItemServiceClient = exports.ExperimentServiceClient = exports.ExperimentArmServiceClient = exports.CustomizerAttributeServiceClient = exports.CustomInterestServiceClient = exports.CustomerUserAccessServiceClient = exports.CustomerUserAccessInvitationServiceClient = exports.CustomerServiceClient = exports.CustomerNegativeCriterionServiceClient = exports.CustomerManagerLinkServiceClient = exports.CustomerLabelServiceClient = exports.CustomerFeedServiceClient = exports.CustomerExtensionSettingServiceClient = exports.CustomerCustomizerServiceClient = exports.CustomerConversionGoalServiceClient = exports.CustomerClientLinkServiceClient = exports.CustomerAssetServiceClient = exports.CustomConversionGoalServiceClient = exports.CustomAudienceServiceClient = exports.ConversionValueRuleSetServiceClient = exports.ConversionValueRuleServiceClient = exports.ConversionUploadServiceClient = void 0;
exports.protos = exports.UserListServiceClient = exports.UserDataServiceClient = exports.ThirdPartyAppAnalyticsLinkServiceClient = exports.SmartCampaignSuggestServiceClient = void 0;
const v11 = require("./v11");
exports.v11 = v11;
const AccountBudgetProposalServiceClient = v11.AccountBudgetProposalServiceClient;
exports.AccountBudgetProposalServiceClient = AccountBudgetProposalServiceClient;
const AccountLinkServiceClient = v11.AccountLinkServiceClient;
exports.AccountLinkServiceClient = AccountLinkServiceClient;
const AdGroupAdLabelServiceClient = v11.AdGroupAdLabelServiceClient;
exports.AdGroupAdLabelServiceClient = AdGroupAdLabelServiceClient;
const AdGroupAdServiceClient = v11.AdGroupAdServiceClient;
exports.AdGroupAdServiceClient = AdGroupAdServiceClient;
const AdGroupAssetServiceClient = v11.AdGroupAssetServiceClient;
exports.AdGroupAssetServiceClient = AdGroupAssetServiceClient;
const AdGroupBidModifierServiceClient = v11.AdGroupBidModifierServiceClient;
exports.AdGroupBidModifierServiceClient = AdGroupBidModifierServiceClient;
const AdGroupCriterionCustomizerServiceClient = v11.AdGroupCriterionCustomizerServiceClient;
exports.AdGroupCriterionCustomizerServiceClient = AdGroupCriterionCustomizerServiceClient;
const AdGroupCriterionLabelServiceClient = v11.AdGroupCriterionLabelServiceClient;
exports.AdGroupCriterionLabelServiceClient = AdGroupCriterionLabelServiceClient;
const AdGroupCriterionServiceClient = v11.AdGroupCriterionServiceClient;
exports.AdGroupCriterionServiceClient = AdGroupCriterionServiceClient;
const AdGroupCustomizerServiceClient = v11.AdGroupCustomizerServiceClient;
exports.AdGroupCustomizerServiceClient = AdGroupCustomizerServiceClient;
const AdGroupExtensionSettingServiceClient = v11.AdGroupExtensionSettingServiceClient;
exports.AdGroupExtensionSettingServiceClient = AdGroupExtensionSettingServiceClient;
const AdGroupFeedServiceClient = v11.AdGroupFeedServiceClient;
exports.AdGroupFeedServiceClient = AdGroupFeedServiceClient;
const AdGroupLabelServiceClient = v11.AdGroupLabelServiceClient;
exports.AdGroupLabelServiceClient = AdGroupLabelServiceClient;
const AdGroupServiceClient = v11.AdGroupServiceClient;
exports.AdGroupServiceClient = AdGroupServiceClient;
const AdParameterServiceClient = v11.AdParameterServiceClient;
exports.AdParameterServiceClient = AdParameterServiceClient;
const AdServiceClient = v11.AdServiceClient;
exports.AdServiceClient = AdServiceClient;
const AssetGroupAssetServiceClient = v11.AssetGroupAssetServiceClient;
exports.AssetGroupAssetServiceClient = AssetGroupAssetServiceClient;
const AssetGroupListingGroupFilterServiceClient = v11.AssetGroupListingGroupFilterServiceClient;
exports.AssetGroupListingGroupFilterServiceClient = AssetGroupListingGroupFilterServiceClient;
const AssetGroupServiceClient = v11.AssetGroupServiceClient;
exports.AssetGroupServiceClient = AssetGroupServiceClient;
const AssetGroupSignalServiceClient = v11.AssetGroupSignalServiceClient;
exports.AssetGroupSignalServiceClient = AssetGroupSignalServiceClient;
const AssetServiceClient = v11.AssetServiceClient;
exports.AssetServiceClient = AssetServiceClient;
const AssetSetAssetServiceClient = v11.AssetSetAssetServiceClient;
exports.AssetSetAssetServiceClient = AssetSetAssetServiceClient;
const AssetSetServiceClient = v11.AssetSetServiceClient;
exports.AssetSetServiceClient = AssetSetServiceClient;
const AudienceInsightsServiceClient = v11.AudienceInsightsServiceClient;
exports.AudienceInsightsServiceClient = AudienceInsightsServiceClient;
const AudienceServiceClient = v11.AudienceServiceClient;
exports.AudienceServiceClient = AudienceServiceClient;
const BatchJobServiceClient = v11.BatchJobServiceClient;
exports.BatchJobServiceClient = BatchJobServiceClient;
const BiddingDataExclusionServiceClient = v11.BiddingDataExclusionServiceClient;
exports.BiddingDataExclusionServiceClient = BiddingDataExclusionServiceClient;
const BiddingSeasonalityAdjustmentServiceClient = v11.BiddingSeasonalityAdjustmentServiceClient;
exports.BiddingSeasonalityAdjustmentServiceClient = BiddingSeasonalityAdjustmentServiceClient;
const BiddingStrategyServiceClient = v11.BiddingStrategyServiceClient;
exports.BiddingStrategyServiceClient = BiddingStrategyServiceClient;
const BillingSetupServiceClient = v11.BillingSetupServiceClient;
exports.BillingSetupServiceClient = BillingSetupServiceClient;
const CampaignAssetServiceClient = v11.CampaignAssetServiceClient;
exports.CampaignAssetServiceClient = CampaignAssetServiceClient;
const CampaignAssetSetServiceClient = v11.CampaignAssetSetServiceClient;
exports.CampaignAssetSetServiceClient = CampaignAssetSetServiceClient;
const CampaignBidModifierServiceClient = v11.CampaignBidModifierServiceClient;
exports.CampaignBidModifierServiceClient = CampaignBidModifierServiceClient;
const CampaignBudgetServiceClient = v11.CampaignBudgetServiceClient;
exports.CampaignBudgetServiceClient = CampaignBudgetServiceClient;
const CampaignConversionGoalServiceClient = v11.CampaignConversionGoalServiceClient;
exports.CampaignConversionGoalServiceClient = CampaignConversionGoalServiceClient;
const CampaignCriterionServiceClient = v11.CampaignCriterionServiceClient;
exports.CampaignCriterionServiceClient = CampaignCriterionServiceClient;
const CampaignCustomizerServiceClient = v11.CampaignCustomizerServiceClient;
exports.CampaignCustomizerServiceClient = CampaignCustomizerServiceClient;
const CampaignDraftServiceClient = v11.CampaignDraftServiceClient;
exports.CampaignDraftServiceClient = CampaignDraftServiceClient;
const CampaignExperimentServiceClient = v11.CampaignExperimentServiceClient;
exports.CampaignExperimentServiceClient = CampaignExperimentServiceClient;
const CampaignExtensionSettingServiceClient = v11.CampaignExtensionSettingServiceClient;
exports.CampaignExtensionSettingServiceClient = CampaignExtensionSettingServiceClient;
const CampaignFeedServiceClient = v11.CampaignFeedServiceClient;
exports.CampaignFeedServiceClient = CampaignFeedServiceClient;
const CampaignGroupServiceClient = v11.CampaignGroupServiceClient;
exports.CampaignGroupServiceClient = CampaignGroupServiceClient;
const CampaignLabelServiceClient = v11.CampaignLabelServiceClient;
exports.CampaignLabelServiceClient = CampaignLabelServiceClient;
const CampaignServiceClient = v11.CampaignServiceClient;
exports.CampaignServiceClient = CampaignServiceClient;
const CampaignSharedSetServiceClient = v11.CampaignSharedSetServiceClient;
exports.CampaignSharedSetServiceClient = CampaignSharedSetServiceClient;
const ConversionActionServiceClient = v11.ConversionActionServiceClient;
exports.ConversionActionServiceClient = ConversionActionServiceClient;
const ConversionAdjustmentUploadServiceClient = v11.ConversionAdjustmentUploadServiceClient;
exports.ConversionAdjustmentUploadServiceClient = ConversionAdjustmentUploadServiceClient;
const ConversionCustomVariableServiceClient = v11.ConversionCustomVariableServiceClient;
exports.ConversionCustomVariableServiceClient = ConversionCustomVariableServiceClient;
const ConversionGoalCampaignConfigServiceClient = v11.ConversionGoalCampaignConfigServiceClient;
exports.ConversionGoalCampaignConfigServiceClient = ConversionGoalCampaignConfigServiceClient;
const ConversionUploadServiceClient = v11.ConversionUploadServiceClient;
exports.ConversionUploadServiceClient = ConversionUploadServiceClient;
const ConversionValueRuleServiceClient = v11.ConversionValueRuleServiceClient;
exports.ConversionValueRuleServiceClient = ConversionValueRuleServiceClient;
const ConversionValueRuleSetServiceClient = v11.ConversionValueRuleSetServiceClient;
exports.ConversionValueRuleSetServiceClient = ConversionValueRuleSetServiceClient;
const CustomAudienceServiceClient = v11.CustomAudienceServiceClient;
exports.CustomAudienceServiceClient = CustomAudienceServiceClient;
const CustomConversionGoalServiceClient = v11.CustomConversionGoalServiceClient;
exports.CustomConversionGoalServiceClient = CustomConversionGoalServiceClient;
const CustomerAssetServiceClient = v11.CustomerAssetServiceClient;
exports.CustomerAssetServiceClient = CustomerAssetServiceClient;
const CustomerClientLinkServiceClient = v11.CustomerClientLinkServiceClient;
exports.CustomerClientLinkServiceClient = CustomerClientLinkServiceClient;
const CustomerConversionGoalServiceClient = v11.CustomerConversionGoalServiceClient;
exports.CustomerConversionGoalServiceClient = CustomerConversionGoalServiceClient;
const CustomerCustomizerServiceClient = v11.CustomerCustomizerServiceClient;
exports.CustomerCustomizerServiceClient = CustomerCustomizerServiceClient;
const CustomerExtensionSettingServiceClient = v11.CustomerExtensionSettingServiceClient;
exports.CustomerExtensionSettingServiceClient = CustomerExtensionSettingServiceClient;
const CustomerFeedServiceClient = v11.CustomerFeedServiceClient;
exports.CustomerFeedServiceClient = CustomerFeedServiceClient;
const CustomerLabelServiceClient = v11.CustomerLabelServiceClient;
exports.CustomerLabelServiceClient = CustomerLabelServiceClient;
const CustomerManagerLinkServiceClient = v11.CustomerManagerLinkServiceClient;
exports.CustomerManagerLinkServiceClient = CustomerManagerLinkServiceClient;
const CustomerNegativeCriterionServiceClient = v11.CustomerNegativeCriterionServiceClient;
exports.CustomerNegativeCriterionServiceClient = CustomerNegativeCriterionServiceClient;
const CustomerServiceClient = v11.CustomerServiceClient;
exports.CustomerServiceClient = CustomerServiceClient;
const CustomerUserAccessInvitationServiceClient = v11.CustomerUserAccessInvitationServiceClient;
exports.CustomerUserAccessInvitationServiceClient = CustomerUserAccessInvitationServiceClient;
const CustomerUserAccessServiceClient = v11.CustomerUserAccessServiceClient;
exports.CustomerUserAccessServiceClient = CustomerUserAccessServiceClient;
const CustomInterestServiceClient = v11.CustomInterestServiceClient;
exports.CustomInterestServiceClient = CustomInterestServiceClient;
const CustomizerAttributeServiceClient = v11.CustomizerAttributeServiceClient;
exports.CustomizerAttributeServiceClient = CustomizerAttributeServiceClient;
const ExperimentArmServiceClient = v11.ExperimentArmServiceClient;
exports.ExperimentArmServiceClient = ExperimentArmServiceClient;
const ExperimentServiceClient = v11.ExperimentServiceClient;
exports.ExperimentServiceClient = ExperimentServiceClient;
const ExtensionFeedItemServiceClient = v11.ExtensionFeedItemServiceClient;
exports.ExtensionFeedItemServiceClient = ExtensionFeedItemServiceClient;
const FeedItemServiceClient = v11.FeedItemServiceClient;
exports.FeedItemServiceClient = FeedItemServiceClient;
const FeedItemSetLinkServiceClient = v11.FeedItemSetLinkServiceClient;
exports.FeedItemSetLinkServiceClient = FeedItemSetLinkServiceClient;
const FeedItemSetServiceClient = v11.FeedItemSetServiceClient;
exports.FeedItemSetServiceClient = FeedItemSetServiceClient;
const FeedItemTargetServiceClient = v11.FeedItemTargetServiceClient;
exports.FeedItemTargetServiceClient = FeedItemTargetServiceClient;
const FeedMappingServiceClient = v11.FeedMappingServiceClient;
exports.FeedMappingServiceClient = FeedMappingServiceClient;
const FeedServiceClient = v11.FeedServiceClient;
exports.FeedServiceClient = FeedServiceClient;
const GeoTargetConstantServiceClient = v11.GeoTargetConstantServiceClient;
exports.GeoTargetConstantServiceClient = GeoTargetConstantServiceClient;
const GoogleAdsFieldServiceClient = v11.GoogleAdsFieldServiceClient;
exports.GoogleAdsFieldServiceClient = GoogleAdsFieldServiceClient;
const GoogleAdsServiceClient = v11.GoogleAdsServiceClient;
exports.GoogleAdsServiceClient = GoogleAdsServiceClient;
const InvoiceServiceClient = v11.InvoiceServiceClient;
exports.InvoiceServiceClient = InvoiceServiceClient;
const KeywordPlanAdGroupKeywordServiceClient = v11.KeywordPlanAdGroupKeywordServiceClient;
exports.KeywordPlanAdGroupKeywordServiceClient = KeywordPlanAdGroupKeywordServiceClient;
const KeywordPlanAdGroupServiceClient = v11.KeywordPlanAdGroupServiceClient;
exports.KeywordPlanAdGroupServiceClient = KeywordPlanAdGroupServiceClient;
const KeywordPlanCampaignKeywordServiceClient = v11.KeywordPlanCampaignKeywordServiceClient;
exports.KeywordPlanCampaignKeywordServiceClient = KeywordPlanCampaignKeywordServiceClient;
const KeywordPlanCampaignServiceClient = v11.KeywordPlanCampaignServiceClient;
exports.KeywordPlanCampaignServiceClient = KeywordPlanCampaignServiceClient;
const KeywordPlanIdeaServiceClient = v11.KeywordPlanIdeaServiceClient;
exports.KeywordPlanIdeaServiceClient = KeywordPlanIdeaServiceClient;
const KeywordPlanServiceClient = v11.KeywordPlanServiceClient;
exports.KeywordPlanServiceClient = KeywordPlanServiceClient;
const KeywordThemeConstantServiceClient = v11.KeywordThemeConstantServiceClient;
exports.KeywordThemeConstantServiceClient = KeywordThemeConstantServiceClient;
const LabelServiceClient = v11.LabelServiceClient;
exports.LabelServiceClient = LabelServiceClient;
const MediaFileServiceClient = v11.MediaFileServiceClient;
exports.MediaFileServiceClient = MediaFileServiceClient;
const MerchantCenterLinkServiceClient = v11.MerchantCenterLinkServiceClient;
exports.MerchantCenterLinkServiceClient = MerchantCenterLinkServiceClient;
const OfflineUserDataJobServiceClient = v11.OfflineUserDataJobServiceClient;
exports.OfflineUserDataJobServiceClient = OfflineUserDataJobServiceClient;
const PaymentsAccountServiceClient = v11.PaymentsAccountServiceClient;
exports.PaymentsAccountServiceClient = PaymentsAccountServiceClient;
const ReachPlanServiceClient = v11.ReachPlanServiceClient;
exports.ReachPlanServiceClient = ReachPlanServiceClient;
const RecommendationServiceClient = v11.RecommendationServiceClient;
exports.RecommendationServiceClient = RecommendationServiceClient;
const RemarketingActionServiceClient = v11.RemarketingActionServiceClient;
exports.RemarketingActionServiceClient = RemarketingActionServiceClient;
const SharedCriterionServiceClient = v11.SharedCriterionServiceClient;
exports.SharedCriterionServiceClient = SharedCriterionServiceClient;
const SharedSetServiceClient = v11.SharedSetServiceClient;
exports.SharedSetServiceClient = SharedSetServiceClient;
const SmartCampaignSettingServiceClient = v11.SmartCampaignSettingServiceClient;
exports.SmartCampaignSettingServiceClient = SmartCampaignSettingServiceClient;
const SmartCampaignSuggestServiceClient = v11.SmartCampaignSuggestServiceClient;
exports.SmartCampaignSuggestServiceClient = SmartCampaignSuggestServiceClient;
const ThirdPartyAppAnalyticsLinkServiceClient = v11.ThirdPartyAppAnalyticsLinkServiceClient;
exports.ThirdPartyAppAnalyticsLinkServiceClient = ThirdPartyAppAnalyticsLinkServiceClient;
const UserDataServiceClient = v11.UserDataServiceClient;
exports.UserDataServiceClient = UserDataServiceClient;
const UserListServiceClient = v11.UserListServiceClient;
exports.UserListServiceClient = UserListServiceClient;
exports.default = { v11, AccountBudgetProposalServiceClient, AccountLinkServiceClient, AdGroupAdLabelServiceClient, AdGroupAdServiceClient, AdGroupAssetServiceClient, AdGroupBidModifierServiceClient, AdGroupCriterionCustomizerServiceClient, AdGroupCriterionLabelServiceClient, AdGroupCriterionServiceClient, AdGroupCustomizerServiceClient, AdGroupExtensionSettingServiceClient, AdGroupFeedServiceClient, AdGroupLabelServiceClient, AdGroupServiceClient, AdParameterServiceClient, AdServiceClient, AssetGroupAssetServiceClient, AssetGroupListingGroupFilterServiceClient, AssetGroupServiceClient, AssetGroupSignalServiceClient, AssetServiceClient, AssetSetAssetServiceClient, AssetSetServiceClient, AudienceInsightsServiceClient, AudienceServiceClient, BatchJobServiceClient, BiddingDataExclusionServiceClient, BiddingSeasonalityAdjustmentServiceClient, BiddingStrategyServiceClient, BillingSetupServiceClient, CampaignAssetServiceClient, CampaignAssetSetServiceClient, CampaignBidModifierServiceClient, CampaignBudgetServiceClient, CampaignConversionGoalServiceClient, CampaignCriterionServiceClient, CampaignCustomizerServiceClient, CampaignDraftServiceClient, CampaignExperimentServiceClient, CampaignExtensionSettingServiceClient, CampaignFeedServiceClient, CampaignGroupServiceClient, CampaignLabelServiceClient, CampaignServiceClient, CampaignSharedSetServiceClient, ConversionActionServiceClient, ConversionAdjustmentUploadServiceClient, ConversionCustomVariableServiceClient, ConversionGoalCampaignConfigServiceClient, ConversionUploadServiceClient, ConversionValueRuleServiceClient, ConversionValueRuleSetServiceClient, CustomAudienceServiceClient, CustomConversionGoalServiceClient, CustomerAssetServiceClient, CustomerClientLinkServiceClient, CustomerConversionGoalServiceClient, CustomerCustomizerServiceClient, CustomerExtensionSettingServiceClient, CustomerFeedServiceClient, CustomerLabelServiceClient, CustomerManagerLinkServiceClient, CustomerNegativeCriterionServiceClient, CustomerServiceClient, CustomerUserAccessInvitationServiceClient, CustomerUserAccessServiceClient, CustomInterestServiceClient, CustomizerAttributeServiceClient, ExperimentArmServiceClient, ExperimentServiceClient, ExtensionFeedItemServiceClient, FeedItemServiceClient, FeedItemSetLinkServiceClient, FeedItemSetServiceClient, FeedItemTargetServiceClient, FeedMappingServiceClient, FeedServiceClient, GeoTargetConstantServiceClient, GoogleAdsFieldServiceClient, GoogleAdsServiceClient, InvoiceServiceClient, KeywordPlanAdGroupKeywordServiceClient, KeywordPlanAdGroupServiceClient, KeywordPlanCampaignKeywordServiceClient, KeywordPlanCampaignServiceClient, KeywordPlanIdeaServiceClient, KeywordPlanServiceClient, KeywordThemeConstantServiceClient, LabelServiceClient, MediaFileServiceClient, MerchantCenterLinkServiceClient, OfflineUserDataJobServiceClient, PaymentsAccountServiceClient, ReachPlanServiceClient, RecommendationServiceClient, RemarketingActionServiceClient, SharedCriterionServiceClient, SharedSetServiceClient, SmartCampaignSettingServiceClient, SmartCampaignSuggestServiceClient, ThirdPartyAppAnalyticsLinkServiceClient, UserDataServiceClient, UserListServiceClient };
const protos = require("../protos/protos");
exports.protos = protos;
